### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by saketkum11**
ProductListing
- filter context
- product context
- sorting 
- reducer

### CODE REVIEW

**Review by shubhamkumar648 — APPROVED** [review]
Good way of using breaking components in smaller components! good use of es6 syntax and naming convention of function name

**Review by Ash-D23 — APPROVED** [review]

File: src/Context/FIlter/Filter-Context.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                return items.sort((a, b) => b.price - a.price);
+            }
+        
+            return items;
+        
+        };
+ 
+   const getFilteredData = (sortedData,fruit,vegetable,eggAndMeat,grain)=>{
+
+    let getFruit = [];
```
Comment (shubhamkumar648) [inline src/Context/FIlter/Filter-Context.js:53]: I like you used to keep every    array in a variable , it helps more readble

File: src/Pages/Home/Home.jsx [hunk 2]
```diff
@@ -2,20 +2,15 @@ import React from "react";
...
+
+
 function Home() {
 return (
 <>
     <Navbar></Navbar>
     <main className="wt-80 m-auto ">
         <div className="flex items-start justify-center h-lg m-y-11 position-rel">
-            <img src={apple} alt="" className="rounded-L object-content wt-100 h-100" />
+            <img src="/Assets/Apple.jpg" alt="no image" className="rounded-L object-content wt-100 h-100" />
```
Comment (shubhamkumar648) [inline src/Pages/Home/Home.jsx:13]: you can also use cloud storage for image

File: src/reducer/filterReducer.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+
+
+const filterReducer = (state,{type,payload}) => {
+
+    const {grain,eggAndMeat,vegetable,fruit} = state.showCategoryInventory;
```
Comment (shubhamkumar648) [inline src/reducer/filterReducer.js:5]: good use object distructuring

**Review by shubhamkumar648 — COMMENTED** [review]
Remiang all functionality  of  code undestandable  , nice way of using context and reducer to avoid prop drilling 
the good variable name,

**Pull request merged**